File: frontend/lib/features/fees/domain/models/monthly_fee_obligation.dart
class MonthlyFeeObligation {
  final String? id;
  final String studentId;
  final String? studentName;
  final int year;
  final int month;
  final double amountDue;
  final double amountPaid;
  final String status;
  final DateTime? dueDate;

  MonthlyFeeObligation({
    this.id,
    required this.studentId,
    this.studentName,
    required this.year,
    required this.month,
    required this.amountDue,
    required this.amountPaid,
    required this.status,
    this.dueDate,
  });

  factory MonthlyFeeObligation.fromJson(Map<String, dynamic> json) {
    return MonthlyFeeObligation(
      id: json['id'],
      studentId: json['studentId'] ?? '',
      studentName: json['studentName'],
      year: json['year'] ?? DateTime.now().year,
      month: json['month'] ?? DateTime.now().month,
      amountDue: (json['amountDue'] ?? 0).toDouble(),
      amountPaid: (json['amountPaid'] ?? 0).toDouble(),
      status: json['status'] ?? 'Unpaid',
      dueDate: json['dueDate'] != null ? DateTime.tryParse(json['dueDate']) : null,
    );
  }
}

File: backend/src/ShriNrityalaya.Domain/Entities/MonthlyFeeObligation.cs
using ShriNrityalaya.Domain.Common;

namespace ShriNrityalaya.Domain.Entities;

public class MonthlyFeeObligation : AuditableEntity
{
    public Guid StudentId { get; set; }
    public Student Student { get; set; } = null!;

    public int Year { get; set; }
    public int Month { get; set; }
    
    public decimal AmountDue { get; set; }
    public decimal AmountPaid { get; set; }
    public string Status { get; set; } = "Unpaid"; // Unpaid, Partial, Paid
    
    public DateTime? DueDate { get; set; }
}
